packages/legend-application-studio/src/stores/editor-state/element-editor-state/service/ServiceExecutionState.ts

```diff
@@ -409,15 +416,18 @@ export class SingleExecutionContextState extends ServiceExecutionContextState {
   }
 
   setMapping(value: Mapping): void {
+    this.executionContext.mapping =
+      PackageableElementExplicitReference.create(value);
     pureSingleExecution_setMapping(
-      this.executionContext,
+      this.executionContext1,
       value,
       this.executionState.editorStore.changeDetectionState.observerContext,
     );
   }
   setRuntime(value: Runtime): void {
+    this.executionContext.runtime = value;
```

Comment:
this line is not necessary, the `pureSingleExecution_setRuntime` sets it already

Reply:
same here as well